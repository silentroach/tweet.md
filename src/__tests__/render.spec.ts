import { readFileSync, readdirSync } from "node:fs";
import { resolve } from "node:path";

import { parse } from "json5";

import { renderTweet } from "../";

const fixturesPath = resolve(__dirname, "__fixtures__");

readdirSync(fixturesPath).forEach((filename) => {
  const filepath = resolve(fixturesPath, filename);
  const data = parse(readFileSync(filepath, { encoding: "utf-8" }));

  it(filename, () => {
    expect(renderTweet(data)).toMatchSnapshot();
  });
});

it("quote above the text", () => {
  const data = parse(
    readFileSync(resolve(fixturesPath, "quote.json5"), { encoding: "utf-8" }),
  );

  expect(
    renderTweet(data, { render: { quotesAboveText: true } }),
  ).toMatchSnapshot();
});
