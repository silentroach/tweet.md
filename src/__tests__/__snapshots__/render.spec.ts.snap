// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ctags.json5 1`] = `"[$APPL](https://twitter.com/search?q=%23APPL&src=ctag) set to jump this morning. General rebound in the market today. Yesterday was a good time to buy into the panic."`;

exports[`emoticon.json5 1`] = `"Testing emoticons 🌞 with ⚽ links [twitter.com](http://t.co/AOZu04FpJG "http://twitter.com")"`;

exports[`emoticon2.json5 1`] = `
"Twitter libraries are a great way to use our API. Did you know:  

✨ 100+ libraries support the Twitter API  
✨ In 25+ programming languages 🐍☕️♦️   

Explore our curated list of community built libraries 📚  

[developer.twitter.com/en/docs/develo…](https://t.co/mgXiWmt9SY "https://developer.twitter.com/en/docs/developer-utilities/twitter-libraries")"
`;

exports[`escaping.json5 1`] = `"Just\\[\\] \\*test \\_the \\(mark&gt;&lt;down\\) \\\\ \\#escaping"`;

exports[`escaping-code.json5 1`] = `"Separate \\\`code escaping\\\` test for future"`;

exports[`escaping-first-number.json5 1`] = `"1234\\. Escape the period after number at the beginning of the string"`;

exports[`escaping-nickname.json5 1`] = `"[@\\_h4\\_](https://twitter.com/_h4_ "Наполеон Бонапарт") done"`;

exports[`extended.json5 1`] = `"some original text"`;

exports[`full-text.json5 1`] = `"some original text"`;

exports[`links.json5 1`] = `"[webpack.github.io/docs/hot-modul…](http://t.co/rYdyv2gX12 "http://webpack.github.io/docs/hot-module-replacement-with-webpack.html"), [webpack.github.io/docs/hot-modul…](http://t.co/32mCbc9Jyr "http://webpack.github.io/docs/hot-module-replacement.html")"`;

exports[`multiline.json5 1`] = `
"Something with multiple  
lines"
`;

exports[`multiple.json5 1`] = `"RT [@miripiruni](https://twitter.com/miripiruni "Slava Oliyanchuk"): [#TIL](https://twitter.com/search?q=%23TIL) [#nodejs](https://twitter.com/search?q=%23nodejs) [pic.twitter.com/VXj36cneyb](http://t.co/VXj36cneyb)"`;

exports[`no-entities.json5 1`] = `"Some text"`;

exports[`quote above the text 1`] = `
"> Some quote

Some text"
`;

exports[`quote.json5 1`] = `
"Some text

> Some quote"
`;

exports[`unknown.json5 1`] = `"Tweet data with some unknown entities will not break the text"`;
